### DISCUSSION TITLE
Change the location of the user data folder (~/.local/share/code-server)

### DISCUSSION THREAD

**Discussion by aghasemi** [coder/code-server#6829] [General] [answered]
Hi,

Is it possible to modify where user data are stored. The configuration of my Kubernetes volumes have some troubles with writeability etc.. in that folder. 

And if so, is it also possible in a Docker image derived from the official `codercom/code-server`?

**Comment by code-asher** [comment, marked as answer]
> Is it possible to modify where user data are stored.

Yes, you can with `--user-data-dir`.  Example: 

```
code-server --user-data-dir /tmp/my-data
```

> is it also possible in a Docker image derived from the official codercom/code-server?

I think it will depend on the image.  With the official image you can just do:

```
docker run codercom/code-server --user-data-dir /tmp/my-data
```

So that might work for unofficial images derived from ours as well.

**Reply by aghasemi** [reply]
Thanks a lot. That worked. However, somehow it seems the heartbeat file (`/home/coder/.local/share/code-server/heartbeat`) is still read from the default location.

**Reply by code-asher** [reply]
Oh!  Let me check...ah yeah you are right, that one is using the XDG directory for data.

I cannot recall if that was on purpose, but changing the default now might be a breaking change.  :thinking: 

One idea is to configure `XDG_DATA_HOME` which should do the trick.

**Reply by code-asher** [reply]
I opened https://github.com/coder/code-server/issues/6834
